lalrpop/src/grammar/parse_tree.rs

```diff
@@ -447,8 +447,8 @@ pub enum SymbolKind {
     // <X>
     Choose(Box<Symbol>),
 
-    // x:X
-    Name(Atom, Box<Symbol>),
+    // <x:X> or <mut x:X>
+    Name(bool, Atom, Box<Symbol>),
```

Comment:
Perhaps an explicit enum would be nice here?

```
enum Mutability {
    Mutable,
    Immutable,
}
```

With a `fmt::Display` impl it could simplify some other code as well

Reply:
I was thinking about that, but I wanted to get it working and get some feedback to see if I missed anything. But before I decide, I want to get the test working or at least understand why it's failing.

Reply:
👍 to an enum